### PR TITLE
ndt: multi stream test

### PR DESCRIPTION
**Pull request by bassosimone**

### CODE REVIEW

File: src/ndt/test_s2c_impl.hpp [hunk 1]
```diff
@@ -41,42 +41,61 @@ void coroutine_impl(Var<Entry> report_entry, std::string address, int port,
...
+                            err = NoError();
+                        }
+                        txp->close([=]() {
+                            ++(*num_completed);
+                            // Note: in this callback we cannot reference
+                            // txp_list or txp because that would keep
+                            // alive txp indefinitely, so we use the num_flows
+                            // variable instead (note that this means that
+                            // the `=` only copies what you use, a thing that
+                            // I was totally unaware of!)
```
Comment (hellais) [inline src/ndt/test_s2c_impl.hpp]: Very good to know this, I also didn't know!

File: src/libmeasurement_kit/cmdline/ndt.cpp [hunk 2]
```diff
@@ -16,13 +16,14 @@ namespace ndt {
...
-    "usage: measurement_kit ndt [-nv] [-C /path/to/ca.bundle] [-p port]\n"
-    "                           [-T download|none|upload] [host]\n";
+  "usage: measurement_kit ndt [-nv] [-C /path/to/ca.bundle] [-p port]\n"
+  "                           [-T download|download-ext|none|upload] [host]\n";
 
 int main(const char *, int argc, char **argv) {
 
     NdtTest test;
     int ch;
+    int test_suite = -1;
```
Comment (bassosimone) [inline src/libmeasurement_kit/cmdline/ndt.cpp]: *fixme*: this should actually be set to zero

**Comment by AntonioLangiu** [comment]
I'm working on it!

File: include/measurement_kit/common/measure_speed.hpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+// information on the copying conditions.
+#ifndef MEASUREMENT_KIT_COMMON_MEASURE_SPEED_HPP
+#define MEASUREMENT_KIT_COMMON_MEASURE_SPEED_HPP
+
+#include <measurement_kit/common/callback.hpp>
+#include <measurement_kit/common/utils.hpp>
+
+namespace mk {
+
+class MeasureSpeed {
```
Comment (AntonioLangiu) [inline include/measurement_kit/common/measure_speed.hpp]: I don't think this class should be here! It's a class used to measure the speed in the NDT test and I thing the correct place would be in the ndt/internal.hpp file, specifically in the Utils section!

File: include/measurement_kit/common/measure_speed.hpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        snap_delay = delay;
+    }
+    MeasureSpeed() {}
+
+    double speed(double ct) const {
+        double elapsed = ct - previous;
+        if (elapsed <= 0.0) {
+            return 0.0;
+        }
+        return (total * 8.0) / 1000.0 / elapsed;
```
Comment (AntonioLangiu) [inline include/measurement_kit/common/measure_speed.hpp]: If this class is meant to be general class used by many tests, maybe different tests want to measure the speed in a different way and not using this function, so I confirm that this class should be in the ndt folder!

**Comment by bassosimone** [comment]
On hold until #899 is merged

**Comment by bassosimone** [comment]
(@AntonioLangiu thanks for committing the merge!)

**Comment by AntonioLangiu** [comment]
I think this is ready to merge! I'm waiting the CI completion and then I will merge it to master!

**Pull request merged**